src/Mutex/Impl/Internal/MutexWorkflow.workflow.cs

```diff
@@ -0,0 +1,55 @@
+using Temporalio.Workflows;
+
+namespace TemporalioSamples.Mutex.Impl.Internal;
+
+[Workflow]
+internal class MutexWorkflow
+{
+    private readonly ILockHandler lockHandler = WorkflowMutex.CreateLockHandler();
+    private readonly Queue<LockRequest> requests = new();
+
+    [WorkflowRun]
+    public async Task RunAsync(MutexWorkflowInput? input = null)
+    {
+        var logger = Workflow.Logger;
+
+        foreach (var request in input?.InitialRequests ?? Array.Empty<LockRequest>())
+        {
+            requests.Enqueue(request);
+        }
+
+        while (!Workflow.ContinueAsNewSuggested)
+        {
+            if (requests.Count == 0)
+            {
+                logger.LogInformation("No lock requests, waiting for more...");
+
+                await Workflow.WaitConditionAsync(() => requests.Count > 0);
+            }
+
+            while (requests.TryDequeue(out var lockRequest))
+            {
+                await this.lockHandler.HandleAsync(lockRequest);
```

Comment:
So it looks like this can only handle one lock at a time which I am not sure is what we want

Reply:
Ah, true, I was thinking of a case where a mutex workflow could lock several independent resources, which doesn't apply here. Can consider this resolved. I think I was confused by the presence of the `ResourceId` in the lock request, which seems like it should be a product of the workflow as a whole.